openedx/core/djangoapps/util/row_delete.py

```diff
@@ -0,0 +1,101 @@
+"""
+Code to delete rows from a table within a Django mgmt command using best practices.
+"""
+
+import logging
+import time
+from django.core.management.base import CommandError, BaseCommand
+from django.db import transaction
+
+log = logging.getLogger(__name__)
+
+
+def delete_rows(
+        model_mgr,
+        table_name,
+        chunk_size,
+        sleep_between
+    ):
+    """
+    Deletes *ALL* rows from table, chunking the deletes to avoid long table/row locks.
+
+    Args:
+        model_mgr (django.db.models.manager.Manager): Django ORM mgr for the table's model.
+        table_name (str): Name of table from which to delete all rows.
+        chunk_size (int): Number of rows to delete in each transaction.
+        sleep_between (float): Number of seconds to sleep between transactions.
+    """
+    if chunk_size <= 0:
+        raise CommandError('Only positive chunk size is allowed ({}).'.format(chunk_size))
+    if sleep_between < 0:
+        raise CommandError('Only non-negative sleep between seconds is allowed ({}).'.format(sleep_between))
+
+    # The "as id" below fools Django raw query into thinking the primary key is being queried.
+    min_max_ids = model_mgr.raw(
+        'SELECT MIN(id) as id, MAX(id) as max_id FROM {}'.format(table_name)
```

Comment:
Something of a nit, but not all tables will have an "id" for the primary key. It would be .0001% more useful to have the column name be passed in.

Reply:
As a matter of fact, the history tables themselves do NOT use `id` as their primary key - they use `history_id` instead. I'll change this code to pass in the name of the primary key column.